src/serac/numerics/functional/domain.hpp

```diff
@@ -135,10 +135,20 @@ struct Domain {
 };
 
 /// @brief constructs a domain from all the elements in a mesh
-Domain EntireDomain(const mfem::Mesh& mesh);
+template <int dim>
+Domain EntireDomain(const mfem::Mesh& mesh)
+{
+  auto tautological_predicate = [](std::vector<tensor<double, dim>>, int) -> bool { return true; };
+  return Domain::ofElements(mesh, tautological_predicate);
+}
 
 /// @brief constructs a domain from all the boundary elements in a mesh
-Domain EntireBoundary(const mfem::Mesh& mesh);
+template <int dim>
+Domain EntireBoundary(const mfem::Mesh& mesh)
+{
+  auto tautological_predicate = [](std::vector<tensor<double, dim>>, int) -> bool { return true; };
+  return Domain::ofBoundaryElements(mesh, tautological_predicate);
+}
```

Comment:
I think these can be implemented in a way that doesn't propagate the template parameter `dim` up to the user, which would make this PR avoid the breaking interface change:

https://godbolt.org/z/c5YeqPrrW